examples/asr/speech_classification/frame_vad_infer.py

```diff
@@ -0,0 +1,249 @@
+            manifest_vad_input=manifest_vad_input,
+            out_dir=cfg.frame_out_dir,
+        )
+        logging.info(f"Finish generating VAD frame level prediction. You can find the prediction in {pred_dir}")
+    else:
+        pred_dir = cfg.frame_out_dir
+
+    frame_length_in_sec = cfg.vad.parameters.shift_length_in_sec
+
+    # overlap smoothing filter
+    if cfg.vad.parameters.smoothing:
+        # Generate predictions with overlapping input segments. Then a smoothing filter is applied to decide the label for a frame spanned by multiple segments.
+        # smoothing_method would be either in majority vote (median) or average (mean)
+        logging.info("Generating predictions with overlapping input segments")
+        smoothing_pred_dir = generate_overlap_vad_seq(
+            frame_pred_dir=pred_dir,
+            smoothing_method=cfg.vad.parameters.smoothing,
+            overlap=cfg.vad.parameters.overlap,
+            window_length_in_sec=cfg.vad.parameters.window_length_in_sec,
+            shift_length_in_sec=cfg.vad.parameters.shift_length_in_sec,
+            num_workers=cfg.num_workers,
+            out_dir=cfg.smoothing_out_dir,
+        )
+        logging.info(
+            f"Finish generating predictions with overlapping input segments with smoothing_method={cfg.vad.parameters.smoothing} and overlap={cfg.vad.parameters.overlap}"
+        )
+        pred_dir = smoothing_pred_dir
+
+    # postprocessing and generate speech segments
+    logging.info("Converting frame level prediction to RTTM files.")
+    rttm_out_dir = generate_vad_segment_table(
+        vad_pred_dir=pred_dir,
+        postprocessing_params=cfg.vad.parameters.postprocessing,
+        frame_length_in_sec=frame_length_in_sec,
+        num_workers=cfg.num_workers,
+        use_rttm=cfg.vad.use_rttm,
+        out_dir=cfg.rttm_out_dir,
+    )
+    logging.info(
+        f"Finish generating speech semgents table with postprocessing_params: {cfg.vad.parameters.postprocessing}"
+    )
+
+    logging.info("Writing VAD output to manifest")
+    key_pred_rttm_map = {}
+    manifest_new = []
+    for entry in manifest_orig:
+        key = Path(entry['audio_filepath']).stem
+        entry['rttm_filepath'] = Path(os.path.join(rttm_out_dir, key + ".rttm")).absolute().as_posix()
+        if not Path(entry['rttm_filepath']).is_file():
+            logging.warning(f"Not able to find {entry['rttm_filepath']} for {entry['audio_filepath']}")
+            entry['rttm_filepath'] = ""
+        manifest_new.append(entry)
+        key_pred_rttm_map[key] = entry['rttm_filepath']
+
+    if not cfg.out_manifest_filepath:
+        out_manifest_filepath = "manifest_vad_output.json"
+    else:
+        out_manifest_filepath = cfg.out_manifest_filepath
+    write_manifest(out_manifest_filepath, manifest_new)
+    logging.info(f"Finished writing VAD output to manifest: {out_manifest_filepath}")
+
+    if cfg.get("evaluate", False):
```

Comment:
I strongly suggest the line from 207 to 245 be a separate function and put it into a new python file called `NeMo/nemo/collections/asr/metrics/det_er.py`

For example, all diarization related metric functions and evaluation functions are in 
https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/metrics/der.py

Reply:
agree. let's try moving the functions to util files to simply the script and for reusability. such as `eval_detection_error`